### PR TITLE
store/tikv: support batch load regions with key range

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve? 

In the implementation of https://github.com/pingcap/ticdc/pull/252 we reuse the TiDB `region_cache` lib. We need a convenient region_cache API to support loading limited regions to the RegionCache within given `startKey` to `endKey`, and returns the loaded regions at the same time.

There are two relevant APIs in TiDB, `LoadRegionsInKeyRange`, `BatchLoadRegionsFromKey`, but they don't meet the requirement. Besides the `LoadRegionsInKeyRange` loads regions one by one, which is less efficient than the `ScanRegions` way.

### What is changed and how it works?

Add a more universal `LoadRegionsInKeyRange` API in region_cache and the `LoadRegionsInKeyRange` and `BatchLoadRegionsFromKey` can reuse this function.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Side effects

Related changes

Release note

### CODE REVIEW

File: store/tikv/region_cache.go [hunk 1]
```diff
@@ -590,36 +591,38 @@ func (c *RegionCache) ListRegionIDsInKeyRange(bo *Backoffer, startKey, endKey []
...
+	var batchRegions []*Region
 	for {
-		curRegion, err := c.loadRegion(bo, startKey, false)
+		batchRegions, err = c.BatchLoadRegionsWithKeyRange(bo, startKey, endKey, defaultRegionsPerBatch)
 		if err != nil {
 			return nil, errors.Trace(err)
 		}
-		c.mu.Lock()
-		c.insertRegionToCache(curRegion)
-		c.mu.Unlock()
```
Comment (lysu) [inline store/tikv/region_cache.go]: `LoadRegionsInKeyRange` will be changed to not refill region cache after this commit?

this method only used in `SHOW TABLE [table_name] REGIONS`, I think it's reasonable to not refill cache

Comment (amyangfei) [reply store/tikv/region_cache.go]: The region cache is still refilled in `BatchLoadRegionsWithKeyRange`, which keeps the same logic before this PR

**Pull request merged**